### ISSUE TITLE
Integrating with OSS-Fuzz

### ISSUE THREAD

**Issue by Google-Autofuzz** [ARM-software/astc-encoder#50]
Greetings astc-encoder developers and contributors,

We’re reaching out because your project is an important part of the open source ecosystem, and we’d like to invite you to integrate with our [fuzzing](https://www.owasp.org/index.php/Fuzzing) service, [OSS-Fuzz]( https://opensource.googleblog.com/2016/12/announcing-oss-fuzz-continuous-fuzzing.html ). OSS-Fuzz is a free fuzzing infrastructure you can use to identify security vulnerabilities and stability bugs in your project, like #36  and #35. OSS-Fuzz will:

- Continuously run all the fuzzers you write.
- Alert you when it finds issues.
- Automatically close issues after they’ve been fixed by a commit.

Many widely used [open source projects]( https://github.com/google/oss-fuzz/tree/master/projects ) like OpenSSL, FFmpeg, LibreOffice, and ImageMagick are fuzzing via OSS-Fuzz, which helps them find and remediate [critical issues]( https://bugs.chromium.org/p/oss-fuzz/issues/list?can=1&q=status%3AFixed%2CVerified+Type%3ABug%2CBug-Security+-component%3AInfra+ ). 

Even though typical integrations can be done in < 100 LoC, we have a [reward program]( https://www.google.com/about/appsecurity/patch-rewards/ ) in place which aims to recognize folks who are not just contributing to open source, but are also working hard to make it more secure.

We want to stress that anyone who meets the eligibility criteria and integrates a project with OSS-Fuzz is eligible for a reward.

If you're not interested in integrating with OSS-Fuzz, it would be helpful for us to understand why—lack of interest, lack of time, or something else—so we can better support projects like yours in the future.

If we’ve missed your question in our [FAQ]( https://github.com/google/oss-fuzz/blob/master/docs/faq.md ), feel free to reply or reach out to us at oss-fuzz-outreach@googlegroups.com.

Thanks!

Julien,
OSS-Fuzz Team

**Comment by alsophian** [comment]
Hi,

I work on one of Google's fuzzing-related teams, and I'm glad to announce that `astcenc` is one of the projects that we're adding to OSS-Fuzz as part of our [open-source internship program](https://blog.google/inside-google/working-google/google-internships-go-virtual-help-open-source/) this summer.  @rjotwani and I have a pull request open at https://github.com/google/oss-fuzz/pull/4188 to add fuzzing support for `astcenc`.  All we'll need from you is an email address (that must be linked to a Google account for [authentication with the OSS-Fuzz issue tracker](https://google.github.io/oss-fuzz/faq/#why-do-you-require-a-google-account-for-authentication)), and once that PR is merged, then you'll start receiving issue reports automatically.  (The old process, where the Google-Autofuzz account would open issues here, requires a Google engineer to work on the issue manually, and is based on the older version of the encoder we use at Google, but OSS-Fuzz continuously checks against the `HEAD` of this repository.)

The OSS-Fuzz documentation is at https://google.github.io/oss-fuzz/, and I'm happy to answer any questions you might have.

Thanks!
Michael

**Comment by solidpixel** [comment]
Thanks for looking at this. It looks like you have it set up with my account already: peter <dot> harris <at> arm <dot> com.

**Comment by solidpixel** [comment]
@alsophian Thanks for getting this first stab at this merged into oss-fuzz. 

As a next step, I'm happy to look at how we can officially host the necessary Makefile modifications and hosting the fuzz targets direct from this repo - it will remove a point of fragility with your team having to maintain patches and fuzz tests that stay in sync with an astcenc checkout.

**Comment by solidpixel** [comment]
Proof-of-concept in-tree fuzzer created in PR #158.
Needs some modifications to the oss-fuzz `build.sh` to pick up the changes, but it can coexist with the fuzzers in-tree in oss-fuzz is we add two build lines.

**Comment by solidpixel** [comment]
Closing this one for now - in-tree build is integrated and working. Will raise separate PRs for looking at new fuzz tests.
